### PR TITLE
Verify C6 signals / compared to ESP-IDF

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Checked some `// TODO: Verify` comments for the signals. Compared/aligned them to ESP-IDF

#### Testing
Just comments removed/edited

### CODE REVIEW

File: esp-hal/src/soc/esp32c6/gpio.rs [hunk 1]
```diff
@@ -341,9 +356,3 @@ impl InterruptStatusRegisterAccess for InterruptStatusRegisterAccessBank0 {
         unsafe { &*GPIO::PTR }.pcpu_nmi_int().read().bits()
     }
 }
-
```
Comment (bjoernQ) [inline esp-hal/src/soc/esp32c6/gpio.rs:344]: C6 doesn't support USB-OTG

File: esp-hal/src/soc/esp32c6/gpio.rs [hunk 2]
```diff
@@ -90,7 +90,14 @@ pub enum InputSignal {
     I2SI_BCK            = 16,
     I2SI_WS             = 17,
     USB_JTAG_TDO_BRIDGE = 19,
-    CPU_TESTBUS0        = 20, // TODO: verify
+    CPU_TESTBUS0        = 20,
```
Comment (SergioGasquez) [inline esp-hal/src/soc/esp32c6/gpio.rs:93]: ESP-IDF source: https://github.com/espressif/esp-idf/blob/636ff35b52f10e1a804a3760a5bd94e68f4b1b71/components/soc/esp32c6/include/soc/gpio_sig_map.h#L43-L50

File: esp-hal/src/soc/esp32c6/gpio.rs [hunk 3]
```diff
@@ -262,7 +277,7 @@ pub enum OutputSignal {
     SPICLK_MUX            = 114,
     SPICS0                = 115,
     SPICS1                = 116,
-    GPIO_TASK_MATRIX_OUT0 = 117, // TODO: verify rhis group - not in TRM but in ESP_IDF
+    GPIO_TASK_MATRIX_OUT0 = 117,
```
Comment (SergioGasquez) [inline esp-hal/src/soc/esp32c6/gpio.rs:280]: ESP-IDF source: https://github.com/espressif/esp-idf/blob/636ff35b52f10e1a804a3760a5bd94e68f4b1b71/components/soc/esp32c6/include/soc/gpio_sig_map.h#L237

**Review by SergioGasquez — APPROVED** [review]
LGTM!

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

**Pull request merged**